fw/connection.h

```diff
@@ -462,8 +469,8 @@ tfw_connection_stop_rcv(TfwConn *conn)
 	 * for HTTP2 client connections, to ensure delivery of
 	 * response with a body larger than HTTP window size.
 	 */
-	return (TFW_CONN_TYPE(conn) & Conn_Stop) &&
-		(!((TFW_CONN_TYPE(conn) & Conn_H2Clnt) == Conn_H2Clnt));
+	return (tfw_connection_was_stopped(conn)
+		&& (!((TFW_CONN_TYPE(conn) & Conn_H2Clnt) == Conn_H2Clnt)));
```

Comment:
`!((TFW_CONN_TYPE(conn) & Conn_H2Clnt) == Conn_H2Clnt)` 
Couldn't it be reduced to `!((TFW_CONN_TYPE(conn) & Conn_H2Clnt)`?

Reply:
No because Conn_H2Clnt = 2 | TFW_FSM_HTTPS, so !((TFW_CONN_TYPE(conn) & Conn_H2Clnt) is false for H2 connections but it also false for https connection. But we should not process anything if error occurs in https connection.  For http2 connection we should do it because we should process window_update frames from the client to have ability to send error response with body

Reply:
Oh, I see, it's not a one-bit flag. Sorry, I should look at it by myself.